src/libstd/sys/common/net.rs

```diff
@@ -42,6 +42,11 @@ use sys::net::netc::IPV6_LEAVE_GROUP as IPV6_DROP_MEMBERSHIP;
               target_os = "solaris", target_os = "haiku")))]
 use sys::net::netc::IPV6_DROP_MEMBERSHIP;
 
+#[cfg(target_os = "linux")]
+use libc::MSG_NOSIGNAL;
```

Comment:
* DragonFlyBSD/NetBSD/OpenBSD/Bitrig:  `MSG_NOSIGNAL = 0x400`;
* FreeBSD ` MSG_NOSIGNAL = 0x20000`;
* Android `MSG_NOSIGNAL = 0x4000`;
* Haiku  `MSG_NOSIGNAL = 0x0800`;

Would be nice if those were included as well.